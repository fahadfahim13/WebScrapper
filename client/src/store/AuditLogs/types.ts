import { Inputs } from 'store/FormInputs/types';

export interface State {
  log: string;
  formItem: Inputs;
}
