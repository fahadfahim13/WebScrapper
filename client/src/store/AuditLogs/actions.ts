import auditLogSlice from './slice';

const { addLog } = auditLogSlice.actions;
export { addLog };
