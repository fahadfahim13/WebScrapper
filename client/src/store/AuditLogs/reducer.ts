import auditLogSlice from './slice';

export default auditLogSlice.reducer;
