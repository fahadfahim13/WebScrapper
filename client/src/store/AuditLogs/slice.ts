import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { State } from './types';

const initialState: State[] = [];

export const auditLogSlice = createSlice({
  name: 'auditLogs',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {
    addLog: (state, action: PayloadAction<State>) => {
      state.push(action.payload);
    },
  },
});

export default auditLogSlice;
