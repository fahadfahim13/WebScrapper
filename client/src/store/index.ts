import { configureStore } from '@reduxjs/toolkit';
import counterReducer from './counter/reducer';
import formReducer from './FormInputs/reducer';
import auditLogsReducer from './AuditLogs/reducer';
import alertReducer from './Alert/reducer';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    formReducer: formReducer,
    auditLogs: auditLogsReducer,
    alertProp: alertReducer,
  },
});
