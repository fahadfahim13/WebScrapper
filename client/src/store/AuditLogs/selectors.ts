import { RootState } from 'store/types';

export const selectAuditLogs = (state: RootState) => state.auditLogs;
