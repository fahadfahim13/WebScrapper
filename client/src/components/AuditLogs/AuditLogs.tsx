import React from 'react';
import { List, ListItem, ListItemText } from '@mui/material';
import {useAppSelector} from 'store/hooks';
import {selectAuditLogs} from 'store/AuditLogs/selectors';

const AuditLogs = () => {
    const auditLogs = useAppSelector(selectAuditLogs);
    return (
        <List>
            {auditLogs.map((log, idx) => (
                <ListItem
                    key={idx}
                    disableGutters
                >
                    <ListItemText secondary={`${log.log}`} />
                </ListItem>
            ))}
        </List>
    );
};

export default AuditLogs;