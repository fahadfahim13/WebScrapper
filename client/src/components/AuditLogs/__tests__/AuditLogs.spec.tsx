import { render, RenderResult } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from 'store';
import AuditLogs from 'components/AuditLogs';

const renderAuditLogs = (): RenderResult => {
    render(
        <Provider store={store}>
            <AuditLogs/>
        </Provider>
    );
};

describe('Audit Logs Snapshot', () => {
    test('Renders Audit logs', () => {
        renderAuditLogs();
        expect(screen).toMatchSnapshot();
    });
})