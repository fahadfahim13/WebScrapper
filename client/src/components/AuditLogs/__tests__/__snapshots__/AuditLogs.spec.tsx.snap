// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit Logs Snapshot Renders Audit logs 1`] = `Screen {}`;
